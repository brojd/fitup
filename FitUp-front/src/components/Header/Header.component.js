import React, { PropTypes } from 'react';
import classNames from 'classnames';
import styles from './Header.stylesheet.css';


class Header extends React.Component {
  constructor() {
    super();
    this._handleScroll = this._handleScroll.bind(this);
    this.state = {
      headerFixed: false
    };
  }
  _handleScroll() {
    if (window.scrollY > 150) {
      this.setState({headerFixed: true});
    } else if (window.scrollY < 3) {
      this.setState({headerFixed: false});
    }
  }
  componentDidMount() {
    window.addEventListener('scroll', this._handleScroll);
  }
  componentWillUnmount() {
    window.removeEventListener('scroll', this._handleScroll);
  }
  render() {
    return (
      <header className={classNames('uk-width-10-10', styles.Header, this.props.className,
                        {[styles.fixedHeader]: this.state.headerFixed}, {'slideInDown': this.state.headerFixed})}>
        <div className='childrenWrapper'>
          { this.props.children }
        </div>
      </header>
    );
  }
}

Header.propTypes = {
  className: PropTypes.string,
  children: React.PropTypes.node
};

export default Header;
