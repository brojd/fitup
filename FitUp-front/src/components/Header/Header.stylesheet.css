.Header {
  border-top: 2px solid #FF1B1B; }

.headerLogo {
  position: absolute;
  z-index: 3;
  left: 50px;
  top: 15px;
  background: #000000; }

.NavBar {
  position: relative; }

.fixedHeader {
  position: fixed;
  z-index: 6;
  box-shadow: 0 2px 5px 0 #350000;
  width: 100vw; }

/*# sourceMappingURL=Header.stylesheet.css.map */
