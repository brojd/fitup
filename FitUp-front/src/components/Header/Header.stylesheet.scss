@import "../../../sass_variables/colors";

.Header {
  border-top: 2px solid $torch_red;
}

.headerLogo {
  position: absolute;
  z-index: 3;
  left: 50px;
  top: 15px;
  background: $black;
}

.NavBar {
  position: relative;
}

.fixedHeader {
  position: fixed;
  z-index: 6;
  box-shadow: 0 2px 5px 0 $temptress;
  width: 100vw;
}
