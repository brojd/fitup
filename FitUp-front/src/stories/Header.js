import React from 'react';
import { storiesOf} from '@kadira/storybook';
import Header from '../components/Header/Header.component';
import headerStyles from '../components/Header/Header.stylesheet.css';
import NavBar from '../components/NavBar/NavBar.component';
import Image from '../components/Image/Image.component';

storiesOf('Header', module)
  .add('header with navbar and logo', () => (
    <Header>
      <Image src='https://s22.postimg.org/hx1k588kx/Logomakr_4x27d_Q.png' alt='logo' height={50} width={68}
             className={headerStyles.headerLogo} />
      <NavBar />
    </Header>
  ));
