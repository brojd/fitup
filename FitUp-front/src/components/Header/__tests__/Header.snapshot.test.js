import React from 'react';
import Header from '../Header.component';
import renderer from 'react-test-renderer';

test('Header renders correctly', () => {
  const component = renderer.create(
    <Header/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
