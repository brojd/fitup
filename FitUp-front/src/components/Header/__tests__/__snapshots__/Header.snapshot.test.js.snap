exports[`test Header renders correctly 1`] = `
<header
  className="uk-width-10-10 Header">
  <div
    className="childrenWrapper" />
</header>
`;
